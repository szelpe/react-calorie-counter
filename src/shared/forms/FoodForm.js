import React, { useContext } from 'react'
import InputField from "./fields/InputField";
import Button from "./fields/Button";
import I18nContext from "../../context/I18nContext";

export default function FoodForm(props) {
    //let { i18n } = useContext(I18nContext);
    let i18n = { t: () => 'Text' };

    return (
        <form onSubmit={handleSubmit}>
            <InputField
                labelText={i18n.t('FoodForm.Food')}
                id="foodName"
                value={props.values.foodName}
                error={props.errors.foodName}
                onChange={props.onFieldChange}
                onBlur={props.onFieldBlur}
            />

            <InputField
                labelText={i18n.t('FoodForm.Calories')}
                type="number"
                id="calorieAmount"
                value={props.values.calorieAmount}
                error={props.errors.calorieAmount}
                onChange={props.onFieldChange}
                onBlur={props.onFieldBlur}
            />

            <Button type="submit">{i18n.t('FoodForm.Add')}</Button>
        </form>
    );

    function handleSubmit(event) {
        event.preventDefault();
        props.onSubmit();
    }
}
