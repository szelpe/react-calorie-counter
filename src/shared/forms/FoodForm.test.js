import renderer from 'react-test-renderer'
import React from 'react'
import FoodForm from "./FoodForm";

test('snapshot test', () => {
    let tree = renderer.create(<FoodForm values={{ foodName: '' }} errors={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});
