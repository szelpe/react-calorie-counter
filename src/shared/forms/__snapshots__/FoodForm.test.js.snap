// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="foodName"
    >
      Food
    </label>
    <input
      className="form-control"
      id="foodName"
      name="foodName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <span
      className="text-danger"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="calorieAmount"
    >
      Calories
    </label>
    <input
      className="form-control"
      id="calorieAmount"
      name="calorieAmount"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
    />
    <span
      className="text-danger"
    />
  </div>
  <div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
